src/graph/query_graph.c

```diff
@@ -10,6 +10,27 @@
 #include "../stores/store.h"
 #include <assert.h>
 
+// Perform DFT on a node to map the connected component it is a part of.
```

Comment:
DFT? don't you mean BFS?

Reply:
This is a preorder recursive DFT. It will use its recursive calls to make arbitrarily-long paths, returning once we've encountered a visited node or can go no further.

A BFS approach would follow all paths from the current node at one time, adding/enqueuing all neighbors and subsequently increasing the depth.